package/base-files/files/bin/config_generate

```diff
@@ -253,6 +254,12 @@ generate_static_system() {
 		add_list system.ntp.server='3.openwrt.pool.ntp.org'
 	EOF
 
+	local labelmac=$(get_mac_label)
+	if [ -n "$labelmac" ]; then
+		set system.@system[-1].hostname="OpenWrt-$(macaddr_geteui $labelmac)"
+	else
+		set system.@system[-1].hostname=OpenWrt
+	fi
```

Comment:
Either remove setting the hostname in the uci batch statement above (line 241), or remove the else statement here.
I'd personally prefer the first option, so to remove from uci batch.